### ISSUE TITLE
Correspond method return nullable type in php7.1~

### ISSUE THREAD

**Issue by chocopie116** [mlively/Phake#242]
# abstract
Corresponded php 7.1 new features
http://php.net/manual/en/migration71.new-features.php#migration71.new-features.nullable-types

# problem
following case

```php
class SomeClass
{
     public function do() :?SomeResult
     {
         if () {
            return null;
         }
         return new SomeResult();
     }
}

class OtherTest extends PHPUnit_Framework_TestCase
{
      public function test_do_return_null()
      {
          $some = Phake::mock('SomeClass');
          Phake::when($some)->do->thenReturn(null); #error triggered
      }
}
```

# error message
```
Return value of SomeClass_PHAKE593fb55222d001::do() must be an instance of SomeResult, null returned
```

**Comment by chocopie116** [comment]
Following Pull Request should be merged (CI on php7.1)
https://github.com/mlively/Phake/pull/241

**Comment by mlively** [comment]
This has been fixed by #247 (just an updated PR from your original #241). Thanks!
